sdk/python/kfp/v2/dsl/type_utils.py

```diff
@@ -57,36 +62,37 @@
     'int': pipeline_spec_pb2.PrimitiveType.INT,
     'double': pipeline_spec_pb2.PrimitiveType.DOUBLE,
     'float': pipeline_spec_pb2.PrimitiveType.DOUBLE,
+    'string': pipeline_spec_pb2.PrimitiveType.STRING,
```

Comment:
Again, IMO we should at least put a deprecation warning on one of them (`str`, `string`) instead of simply aliasing, if not directly failing

Reply:
Then perhaps also do the same for other primitive types for consistency. Also consider aliasing `Text` as well.

> It means we temporarily allow this behavior but it will change eventually.

I'm in favor of relying on explicit annotation instead of inference based on name. So I'll say, yes, this should be changed eventually. WDYT? And I believe if we make this more explicit, we can get rid of this dilemma.

Reply:
JFYI:
When using `create_component_from_func`, the type names are canonicalized when the user uses python types. But the user could technically still use the `"str"` type name in python or component.yaml. Treating them the same might be nice.